File: 547-number-of-provinces/number-of-provinces.cpp
class Solution {
private:
    void dfs(int node, vector<vector<int>>& isConnected, vector<int>& visited) {
        int n = isConnected.size();
        visited[node] = 1;
        for (int j = 0; j < n; j++) {
            if (isConnected[node][j] == 1 && !visited[j]) {
                dfs(j, isConnected, visited);
            }
        }
    }
public:
    int findCircleNum(vector<vector<int>>& isConnected) {
        int count = 0;
        int n = isConnected.size();
        vector<int> visited(n, 0);
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                count++;
                dfs(i, isConnected, visited);
            }
        }
        return count;
    }
};

File: 547-number-of-provinces/number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        n=len(isConnected)
        count=0
        vis=set()
        for i in range(n):
            if i not in vis:
                count+=1
                queue=deque([i])
                vis.add(i)
                while queue:
                    city=queue.pop()
                    for nei in range(n):
                        if isConnected[city][nei]==1 and nei not in vis:
                            vis.add(nei)
                            queue.append(nei)


        return count